src/Features/Core/Portable/Completion/CompletionTrigger.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    /// <summary>
+    /// The action that triggered completion to start.
+    /// </summary>
+    public struct CompletionTrigger
+    {
+        /// <summary>
+        /// The reason that completion was started.
+        /// </summary>
+        public CompletionTriggerKind Kind { get; }
+
+        /// <summary>
+        /// The character associated with the triggering action.
+        /// </summary>
+        public char Character { get; }
```

Comment:
Being non-null implies to me as the reader that it's always there. I'd prefer a nullable.